client/components/product.js

```diff
@@ -24,9 +24,13 @@ export class productComponent extends Component {
           <p>Description: {product && product.description}</p>
           <form onSubmit={async (event) => {
             event.persist();
-            let { cartId } = this.props;
-            if (!this.props.cartId) {
-              const newCart = await this.props.handleCreateCart(event, this.props.user.id);
+            let { cartId, user } = this.props;
+            if (Object.keys(user).length === 0) { //SO suggests && obj.constructor === Object
```

Comment:
The reason is because of `Date`. If you do this you'll get `true`.

```
const x = new Date();
Object.keys(Date).length === 0
```

So checking `obj.constructor` to be equal to `Object` is checking that the constructor function is `Object`.

Reply:
It's a nice check though. If you use `PropTypes` I'm pretty sure React will do a bit of type checking.